src/openapi_mutator/mod.rs

```diff
@@ -226,18 +229,21 @@ fn mutate_number<S: HasRand>(state: &mut S, n: &mut serde_json::value::Number) -
     if let Some(x) = n.as_u64() {
         *n = match x as usize {
             0 => 0.into(),
-            x_usz => state.rand_mut().below(x_usz.saturating_mul(2)).into(),
+            x_usz => state
+                .rand_mut()
+                .below(core::num::NonZero::new(x_usz.saturating_mul(2)).unwrap())
+                .into(),
         };
         return MutationResult::Mutated;
     };
     if let Some(x) = n.as_i64() {
         // always negative
         *n = (state.rand_mut().below(
-            x.checked_neg()
-                .unwrap_or(i64::MAX) // because -i64::MIN == i64::MIN
-                .saturating_mul(4)
+            core::num::NonZero::new(x)
+                .unwrap_or(core::num::NonZero::new(i64::MAX).unwrap()) // because -i64::MIN == i64::MIN
+                .saturating_mul(core::num::NonZero::new(4).unwrap())
```

Comment:
Also this is ugly, any suggestions?

Reply:
The checked_neg disappeared, is that intentional?

Reply:
I suggest importing `core::num::NonZero` since you use it often, and to do the conversion after computing `-4x` since its kernel seems to be just 0 :)

Reply:
Like so: 281c69590d8b34b818907803a01bc494330e6477?